hoomd/md/pytest/test_custom_force.py

```diff
@@ -0,0 +1,326 @@
+import pytest
+import numpy as np
+import numpy.testing as npt
+
+try:
+    import cupy
+    CUPY_IMPORTED = True
+except ImportError:
+    CUPY_IMPORTED = False
+
+import hoomd
+from hoomd import md
+
+
+class MyForceCPU(md.force.Custom):
+
+    def __init__(self):
+        super().__init__()
+
+    def set_forces(self, timestep):
+        with self.cpu_local_force_arrays as arrays:
+            arrays.force = -5
+            arrays.potential_energy = 37
+            arrays.torque = 23
+            arrays.virial = np.arange(6)[None, :]
+
+
+class MyForceGPU(md.force.Custom):
+
+    def __init__(self):
+        super().__init__()
+
+    def set_forces(self, timestep):
+        with self.gpu_local_force_arrays as arrays:
+            arrays.force = -5
```

Comment:
How does this work? I thought we needed the `[:]` syntax to write to the elements of the array?

Reply:
Should we update the global snapshots to use the same shorthand?

Reply:
I am fine with either switch. Where setting directly is an error or setting directly is allowed.

Reply:
I am a fan of allowing bracketless notation to work, but not advertising it. I can update examples and unit tests to use the bracket notation if need be.

If we enforce the bracket notation, we need to raise errors when users use the bracketless notation so there aren't silent issues with user code.